src/itemlist_formaction.cpp

```diff
@@ -361,6 +361,21 @@ void itemlist_formaction::process_operation(operation op, bool automatic, std::v
 			v->show_error(strprintf::fmt(_("Error: couldn't mark feed read: %s"), e.what()));
 		}
 		break;
+	case OP_MARKFEEDREAD_UNTIL_ARTICLE: 
+		LOG(level::INFO, "itemlist_formaction: marking feed read until current article");
+		v->set_status(_("Marking feed read until current article..."));   
+		if (itemposname.length() > 0 && itempos < visible_items.size()) {
+			for (unsigned int i=0; i<=itempos&&i<visible_items.size(); ++i) {
```

Comment:
I don't think it's necessary to check that `i < visible_items.size()`—the code already checks that `i <= itempos`, and it ensured that `itempos < visible_items.size()` earlier. In short, we already know that `i < visible_items.size()`.

I also think it should say `i < itempos`, because both "until" and "above" are both non-inclusive. Is there any reason to use `<=` here?

Reply:
I agree with t `i < visible_items.size()` it's not needed. I will remove that. But I don't agree (not sure) about <= because after few days of actual usage of the feature I noticed that <= is useful. Because I normally scroll the articles or even read current articile then pressing the button for action ("Mark all above as read") marks everything above and the current article as read. It normally means that current article was read. So it's OK that it was marked as read too. I found useful. While I am not sure. I may use `<` instead of `<=` it does not much matter for me. 
And yes, I was in one situation: I was in a feed with just one unread article (which was current article). So I automatically pressed 'a' and it was marked as read. Which was OK for me. So I don't know.

Reply:
BTW how does the command "Mark all above as read" works on iPhone - does it mark current article as read or not?

Reply:
Re `<=`: if we keep it, we'll need to come up with a new name again. "Mark as read from here and up"? Sounds meh.

I'm told that on iPhone it doesn't mark the current article.